src/common/ant.ts

```diff
@@ -76,6 +86,13 @@ export class AoANTReadable implements AoANTRead {
     const res = await this.process.read<AoANTState>({
       tags,
     });
+    const schemaResult = AntStateSchema.safeParse(res);
```

Comment:
Looking at this I wonder if we want to be able provide a `schema` and `errorSchema` to process.read and process.write to enable tapping into the configured logger as well as configuring behaviour on bad results.

We have some specific patterns on error in the ANT and ArNS processes that might be good to enforce.

Reply:
i'd prefer this over providing it to the constructor, but it pushes down the error handling to a class that we should avoid over prescribing. I'd rather these exist on the higher level classes `ANT` and `IO` so we can responsibly handle separately from AO specific errors